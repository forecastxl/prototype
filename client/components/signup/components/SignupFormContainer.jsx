import React, { Component, PropTypes } from 'react'
import { connect } from 'react-redux'
import { requestAccount } from '../actions'
import SignupForm from './SignupForm'

export class SignupFormContainer extends Component {
  constructor(props) {
    super(props)
    this.state = {
      account: {
        firstName: '',
        lastName: '',
        companyName: '',
        phoneNumber: '',
        email: '',
        password: '',
        passwordConfirmation: '',
        agreedToGeneralTerms: false
      }
    }
    this.changeAccountText = this.changeAccountText.bind(this)
    this.changeAccountCheckbox = this.changeAccountCheckbox.bind(this)
    this.processForm = this.processForm.bind(this)
  }

  changeAccountText(event) {
    this.setState({
      account: {
        ...this.state.account,
        [event.target.name]: event.target.value
      }
    })
  }

  changeAccountCheckbox(event) {
    this.setState({
      account: {
        ...this.state.account,
        [event.target.name]: event.target.checked
      }
    })
  }

  processForm(event) {
    event.preventDefault()
    this.props.requestAccount(this.state.account)
  }

  render() {
    return (
      <SignupForm
        errors={this.props.errors}
        onTextChange={this.changeAccountText}
        onCheckboxChange={this.changeAccountCheckbox}
        onSubmit={this.processForm}
        account={this.state.account}
      />
    )
  }
}

SignupFormContainer.propTypes = {
  errors: PropTypes.object.isRequired,
  requestAccount: PropTypes.func.isRequired
}

const mapStateToProps = state => ({
  errors: state.signup.errors
})

export default connect(mapStateToProps, { requestAccount })(SignupFormContainer)
