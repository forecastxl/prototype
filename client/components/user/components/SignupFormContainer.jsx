import React, { Component, PropTypes } from 'react'
import { connect } from 'react-redux'
import * as types from '../actionTypes'
import { createAccount } from '../actions'
import SignupForm from './SignupForm'

export class SignupFormContainer extends Component {
  constructor(props) {
    super(props)
    this.state = {
      account: {
        firstName: '',
        lastName: '',
        companyName: '',
        phoneNumber: '',
        email: '',
        password: '',
        passwordConfirmation: '',
        agreedToGeneralTerms: false
      }
    }
    this.changeAccountText = this.changeAccountText.bind(this)
    this.changeAccountCheckbox = this.changeAccountCheckbox.bind(this)
    this.processForm = this.processForm.bind(this)
  }

  changeAccountText(event) {
    this.setState({
      account: {
        ...this.state.account,
        [event.target.name]: event.target.value
      }
    })
  }

  changeAccountCheckbox(event) {
    this.setState({
      account: {
        ...this.state.account,
        [event.target.name]: event.target.checked
      }
    })
  }

  processForm(event) {
    event.preventDefault()
    this.props.createAccount(this.state.account)
  }

  render() {
    return (
      <SignupForm
        errors={this.props.errors}
        onTextChange={this.changeAccountText}
        onCheckboxChange={this.changeAccountCheckbox}
        onSubmit={this.processForm}
        account={this.state.account}
      />
    )
  }
}

SignupFormContainer.propTypes = {
  errors: PropTypes.object.isRequired,
  createAccount: PropTypes.func.isRequired
}

const mapStateToProps = state => ({
  errors: state.api[types.CREATE_ACCOUNT].errors
})

export default connect(mapStateToProps, { createAccount })(SignupFormContainer)
