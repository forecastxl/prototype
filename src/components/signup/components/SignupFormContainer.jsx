import React, { Component, PropTypes } from 'react'
import { connect } from 'react-redux'
import { requestAccount } from '../actions'
import SignupForm from './SignupForm'

export class SignupFormContainer extends Component {
  constructor(props) {
    super(props)
    this.state = {
      account: {
        firstName: '',
        lastName: '',
        companyName: '',
        phoneNumber: '',
        locale: '',
        email: '',
        password: '',
        passwordConfirmation: '',
        agreedToGeneralTerms: ''
      }
    }
    this.changeAccount = this.changeAccount.bind(this)
    this.processForm = this.processForm.bind(this)
  }

  changeAccount(event) {
    this.setState({
      account: {
        ...this.state.account,
        [event.target.name]: event.target.value
      }
    })
  }

  processForm(event) {
    event.preventDefault()
    this.props.requestAccount(this.state.account)
  }

  render() {
    return (
      <SignupForm
        errors={this.props.errors}
        onChange={this.changeAccount}
        onSubmit={this.processForm}
        account={this.state.account}
      />
    )
  }
}

SignupFormContainer.propTypes = {
  errors: PropTypes.object.isRequired,
  requestAccount: PropTypes.func.isRequired
}

const mapStateToProps = state => ({
  errors: state.session.errors
})

export default connect(mapStateToProps, { requestAccount })(SignupFormContainer)
