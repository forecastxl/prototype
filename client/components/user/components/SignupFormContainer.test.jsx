import React from 'react'
import { shallow, mount } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import { SignupFormContainer } from './SignupFormContainer'

jest.mock('material-ui/Card')
jest.mock('material-ui/TextField')
jest.mock('material-ui/RaisedButton')
jest.mock('material-ui/Checkbox')

describe('<SignupFormContainer />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <SignupFormContainer
        errors={{}}
        createAccount={() => {}}
      />
    )
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('updates account state text', () => {
    const wrapper = shallow(
      <SignupFormContainer
        errors={{}}
        createAccount={() => {}}
      />
    )
    const event = {
      target: {
        name: 'password',
        value: 'updated'
      }
    }
    const expected = {
      ...wrapper.state('account'),
      [event.target.name]: event.target.value
    }

    wrapper.instance().changeAccountText(event)
    expect(wrapper.state('account')).toEqual(expected)
  })

  it('updates account state checkbox', () => {
    const wrapper = shallow(
      <SignupFormContainer
        errors={{}}
        createAccount={() => {}}
      />
    )
    const event = {
      target: {
        name: 'agreedToGeneralTerms',
        checked: true
      }
    }
    const expected = {
      ...wrapper.state('account'),
      [event.target.name]: event.target.checked
    }

    wrapper.instance().changeAccountCheckbox(event)
    expect(wrapper.state('account')).toEqual(expected)
  })

  it('processes logins', () => {
    const wrapper = mount(
      <SignupFormContainer
        errors={{}}
        createAccount={jest.fn()}
      />
    )
    const event = {
      preventDefault: jest.fn()
    }

    wrapper.instance().processForm(event)
    expect(event.preventDefault).toHaveBeenCalled()
    expect(wrapper.prop('createAccount')).toHaveBeenCalledWith(wrapper.state('account'))
  })
})
