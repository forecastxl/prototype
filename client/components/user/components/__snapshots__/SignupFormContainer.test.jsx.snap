// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignupFormContainer /> renders correctly 1`] = `
<LoginForm
  account={
    Object {
      "agreedToGeneralTerms": false,
      "companyName": "",
      "email": "",
      "firstName": "",
      "lastName": "",
      "password": "",
      "passwordConfirmation": "",
      "phoneNumber": "",
    }
  }
  errors={Object {}}
  onCheckboxChange={[Function]}
  onSubmit={[Function]}
  onTextChange={[Function]}
/>
`;
